### DISCUSSION TITLE
zero copy between ByteBuf(netty) and protobuf

### DISCUSSION THREAD

**Discussion by LinHuiG** [netty/netty#13808] [Q&A]
we used protobuf in netty,it powerful but we need read bytes from ByteBuf before decode or encode to bytes and then write to ByteBuf ,this once copy between ByteBuf(netty) and protobuf make me puzzle.
i found method writeTo and readFrom in protoBuf,but i dont konw how to use it in netty to achieve zero copy,anyone can tell me ?
very thanks!
`ByteBuf::nioBuffer()` cant work, one ByteBuf could has multiple ByteBuffer,i think we need nioBuffers() if we want to get ByteBuffer  from ByteBuf
(i dont konw why the previous discussion was closed,so I opened a new one)

**Comment by lhotari** [comment]
https://github.com/splunk/lightproto supports this.
